common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AADBrokerAuthorizationRequest.java

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory;
+
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.exception.ClientException;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+public class AADBrokerAuthorizationRequest extends AzureActiveDirectoryAuthorizationRequest {

```

Comment:
`AadBrokerAuthorizationRequest` - per treat-acronyms-as-words.

As a side note, `git` [can be dumb about file renames where only capitalization changes](https://stackoverflow.com/questions/10523849/changing-capitalization-of-filenames-in-git) so mind what's committed

Reply:
Consider `AzureActiveDirectoryBrokerAuthorizationRequest` -- this name is long, but is consistent with the form of its superclass (and other AAD supporting classes)